gptme/config.py

```diff
@@ -205,16 +164,144 @@ def get_project_config(workspace: Path | None) -> ProjectConfig | None:
         )
         # load project config
         with open(project_config_path) as f:
-            config_data = dict(tomlkit.load(f))
+            config_data = tomlkit.load(f)
 
         # Handle RAG config conversion before creating ProjectConfig
         if "rag" in config_data:
             config_data["rag"] = RagConfig(**config_data["rag"])  # type: ignore
 
-        return ProjectConfig(**config_data)  # type: ignore
+        mcp = config_data.pop("mcp", {})
+        mcp_config = MCPConfig.from_toml(mcp)
+
+        return ProjectConfig(**config_data, mcp=mcp_config)  # type: ignore
```

Comment:
Ah, this is the kind of thing that gave us Table's instead of dicts in the config objects, wrecking `dataclasses.asdict`.

Reply:
Fixed